### ISSUE TITLE
Grid Multisort

### ISSUE THREAD

**Issue by clivemoj** [vaadin/flow-components#1197]
Version: Vaadin Flow 14.1.2
Component: Grid

User sorting on several columns works back to front.
Latest column clicked becomes the no 1 sorting column, which is very counter-intuitive.
This is true even when the user just change the sorting direction.

Let's say you have a list of people with FIRST and LAST name and the user decides that he wish to sort by LAST name and then FIRST name, which is a pretty normal situation.
He is then forced to first click FIRST and then LAST, the opposite to what he was looking to do!
And, when he got the hang of that he changes his mind and want the FIRST name in descending order and clicks FIRST name again, then that column not only changes direction but also becomes the no 1 sorting column.

This needs to be fixed or possible to control via settings.

PS.
We are very happy about the improvement that allows the user to remove sorting.
DS.

**Comment by KasparScherrer** [comment]
Hi, I very much agree with clivemoj here. This is very counter-intuitive!

The first column to be sorted by the user should ALWAYS remain the first sorting until the user removes the same again. Each subsequently applied column-sorting should not be priorized to the first one. 

When a user wants to sort a Grid by lastName, then firstName, then age, the user instinctly clicks on lastName, then firstName, then age. Because every sortable Grid would behave this way. But in a Vaadin Application, this would now lead to a sorting of age, then firstName, then lastName. Very strange behaviour! Even (or especially) computer-savy users have reported problems with grid sorting in my applications.

I, too, would love to see this fixed or at least controllable with a setting/configuration.

**Comment by skiedrowski** [comment]
this is still true using Vaadin 21.0.2 - and it is very confusing

**Comment by mictaege** [comment]
In my opinion the current behavior is a showstopper. If multisort would work the intuitive way, it would be a great and usefull feature, but currently we don't offer it to our users at all. 

Are there any plans to change?

**Comment by MatthiasVanMeerbeek** [comment]
Compared to Vaadin 8 the sorting works different and i.m.o. also counter-intuitive but some might find the new way intuitive. Providing an option to choose how you want to have the sorting is probably the optimal way to make both parties happy. 

In Vaadin 8 you just had sort by ascending or descending order but in Vaadin Flow there is also a "default" or "neutral" state/order. If you use MultiSort you can just click on the headers to add (ascending/descending) or remove (default/neutral) sorting on the specific column. In the past you had to shift-click the header of the columns to to be able to use the MultiSort. It would be nice if the MultiSort was again only available when using shift-click. When you f.e. have 5 columns sorted right now you need to go through all the columns to set them back to the "default"/"neutral" state instead of just left clicking on one column header. This is also not so user-friendly i.m.o. 

In the new (counter-intuitive) way it is also a lot harder to f.e. change the sorting (f.e. change from ascending to descending) for a specific column. 
Example: There are two columns, column A and column B. Column A is sorted in ascending order and is nr.1 and column B is sorted in descending order and is nr.2. I want to change column B to descending order but I want column A to stay nr.1. I can't just click on B because that would make B nr.1. I would have to click on B (and set it to descending) and then I would also have to click on A again (and set it back to ascending). 
So basically clicking on a header currently not only changes the sorting but also changes its "priority" if I you could call it that way.

**Comment by neelu-sarkar** [comment]
Adding my +1 for this issue

**Comment by stefanuebe** [comment]
+1

**Comment by web-padawan** [comment]
Related issue in the web component: https://github.com/vaadin/web-components/issues/1281

**Comment by quartelh** [comment]
+1
